test/benchmarks/datadog/metric_handler.py

```diff
@@ -0,0 +1,167 @@
+from enum import Enum
+import os
+from time import time
+from typing import Dict, List, Optional
+
+import datadog
+from tenacity import retry, retry_if_exception, stop_after_attempt, wait_fixed
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Tag(Enum):
+    @classmethod
+    def values(cls):
+        return [e.value for e in cls]
+
+
+class NoneTag(Tag):
+    none = "none_none_none_none-1234"  # should not match any other tag
+
+
+class DatasetSizeTags(Tag):
+    size_100k = "dataset_size:100k"
+
+
+class ReaderModelTags(Tag):
+    debertabase = "reader:debertabase"
+    debertalarge = "reader:debertalarge"
+    tinyroberta = "reader:tinyroberta"
+
+
+class RetrieverModelTags(Tag):
+    bm25 = "retriever:bm25"
+    minilm = "retriever:minilm"
+    mpnetbase = "retriever:mpnetbase"
+
+
+class DocumentStoreModelTags(Tag):
+    opensearch = "documentstore:opensearch"
+    elasticsearch = "documentstore:elasticsearch"
+    weaviate = "documentstore:weaviate"
+
+
+class BenchmarkType(Tag):
+    retriever = "benchmark_type:retriever"
+    retriever_reader = "benchmark_type:retriever_reader"
+    reader = "benchmark_type:reader"
+
+
+class CustomDatadogMetric:
+    name: str
+    timestamp: float
+    value: float
+    tags: List[Tag]
+
+    def __init__(self, name: str, value: float, tags: Optional[List[Tag]] = None) -> None:
+        self.timestamp = time()
+        self.name = name
+        self.value = value
+        self.tags = self.validate_tags(tags)
+
+    def validate_tags(self, tags: Optional[List[Tag]]) -> List[Tag]:
+        valid_tags = []
+        if tags is not None:
+            for tag in tags:
+                is_dataset_size_tag = tag.value in DatasetSizeTags.values()
```

Comment:
What's the point of using enums if then we ignore the `Tag` type and just look for tag strings?